src/pages/iou/request/step/IOURequestStepWaypoint.js

```diff
@@ -0,0 +1,278 @@
+    const {isOffline} = useNetwork();
+    const textInput = useRef(null);
+    const parsedWaypointIndex = parseInt(pageIndex, 10);
+    const allWaypoints = lodashGet(transaction, 'comment.waypoints', {});
+    const currentWaypoint = lodashGet(allWaypoints, `waypoint${pageIndex}`, {});
+
+    const waypointCount = _.size(allWaypoints);
+    const filledWaypointCount = _.size(_.filter(allWaypoints, (waypoint) => !_.isEmpty(waypoint)));
+
+    const waypointDescriptionKey = useMemo(() => {
+        switch (parsedWaypointIndex) {
+            case 0:
+                return 'distance.waypointDescription.start';
+            case waypointCount - 1:
+                return 'distance.waypointDescription.finish';
+            default:
+                return 'distance.waypointDescription.stop';
+        }
+    }, [parsedWaypointIndex, waypointCount]);
+
+    const waypointAddress = lodashGet(currentWaypoint, 'address', '');
+    // Hide the menu when there is only start and finish waypoint
+    const shouldShowThreeDotsButton = waypointCount > 2;
+    const shouldDisableEditor =
+        isFocused &&
+        (Number.isNaN(parsedWaypointIndex) || parsedWaypointIndex < 0 || parsedWaypointIndex > waypointCount || (filledWaypointCount < 2 && parsedWaypointIndex >= waypointCount));
+
+    const validate = (values) => {
+        const errors = {};
+        const waypointValue = values[`waypoint${pageIndex}`] || '';
+        if (isOffline && waypointValue !== '' && !ValidationUtils.isValidAddress(waypointValue)) {
+            ErrorUtils.addErrorMessage(errors, `waypoint${pageIndex}`, 'bankAccount.error.address');
+        }
+
+        // If the user is online, and they are trying to save a value without using the autocomplete, show an error message instructing them to use a selected address instead.
+        // That enables us to save the address with coordinates when it is selected
+        if (!isOffline && waypointValue !== '' && waypointAddress !== waypointValue) {
+            ErrorUtils.addErrorMessage(errors, `waypoint${pageIndex}`, 'distance.errors.selectSuggestedAddress');
+        }
+
+        return errors;
+    };
+
+    const saveWaypoint = (waypoint) => Transaction.saveWaypoint(transactionID, pageIndex, waypoint, false);
+
+    const submit = (values) => {
+        const waypointValue = values[`waypoint${pageIndex}`] || '';
+
+        // Allows letting you set a waypoint to an empty value
+        if (waypointValue === '') {
+            Transaction.removeWaypoint(transaction, pageIndex, true);
+        }
+
+        // While the user is offline, the auto-complete address search will not work
+        // Therefore, we're going to save the waypoint as just the address, and the lat/long will be filled in on the backend
+        if (isOffline && waypointValue) {
+            const waypoint = {
+                lat: null,
+                lng: null,
+                address: waypointValue,
+                name: values.name,
```

Comment:
Coming from https://github.com/Expensify/App/issues/33505

We didn't provide a fallback value, as we [did in the "old flow"](https://github.com/Expensify/App/pull/31898/files).